src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -781,7 +781,7 @@ private static unsafe void Callback(void* configurationContext, void* name, void
 
                 case GCConfigurationType.StringUtf8:
                     {
-                        string? dataAsString = Marshal.PtrToStringUTF8((IntPtr)data);
+                        string? dataAsString = Marshal.PtrToStringUTF8(checked((IntPtr)data));
```

Comment:
This is `long data`, so conversion is correct.

Seems a bit odd to me that we wouldn't have `GCInterface_EnumerateConfigurationValues` take `void*` or `nint`, however. Maybe a place we can fixup the native code.

Reply:
What happens if this throws? In general don't we try to avoid exceptions thrown across an interop boundary? Does it happen to "just work" in this particular case?

Reply:
The data argument can be either 64-bit integer, a pointer to utf8 zero-terminated string, or a `bool` value.

Reply:
For this branch it's a pointer, though, yes?  If it's not, we have bigger issues than it overflowing, since we're about to dereference it and read it as a string.  I'd assume the checked here is superfluous.

Reply:
Yes, it is a pointer for this branch.

> I'd assume the checked here is superfluous.

Correct.

Reply:
Casting to `nint` instead

Reply:
I am wondering whether it would make sense to introduce overloads for PtrToStringUTF8 and similar APIs that take `void*` to avoid `nint` casts in these situations.

Reply:
I think it'd make sense to do that most anywhere we have an API taking an IntPtr that's to be used like a pointer.

For .NET 8, though.

Reply:
> For .NET 8, though.

Agree.

I do not think that the changes in this PR should go into .NET 7. I do not see any actual problems that it is fixing. It just has a potential to introduce regressions.

Reply:
> I do not think that the changes in this PR should go into .NET 7. I do not see any actual problems that it is fixing. It just has a potential to introduce regressions.

I think that's fine as well.  And this makes me nervous about the analyzer; we should strongly consider having it be info level by default rather than warning level by default.

Reply:
Filed an issue https://github.com/dotnet/runtime/issues/75630